From 74fef0a9973e62df16ff8fc97a795bce1fa2a273 Mon Sep 17 00:00:00 2001
From: David Edmundson <kde@davidedmundson.co.uk>
Date: Tue, 16 Feb 2021 11:27:53 +0000
Subject: [PATCH] [libkworkspace] Interim fix for the logout issue

Calls to a DBus activated service can fail if the sender quits whilst
the service is spawning
(https://gitlab.freedesktop.org/dbus/dbus/-/issues/72) and using
dbus-daemon.

This is a lazy interim fix that just makes these calls block as proper
fixes will require more work.

BUG: 432460


(cherry picked from commit 81d61861608012e4d7a19e6f85b8d136c298f31f)
---
 libkworkspace/sessionmanagement.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/libkworkspace/sessionmanagement.cpp b/libkworkspace/sessionmanagement.cpp
index 4b80485cb..3dbe40f0d 100644
--- a/libkworkspace/sessionmanagement.cpp
+++ b/libkworkspace/sessionmanagement.cpp
@@ -130,10 +130,10 @@ void SessionManagement::requestShutdown(ConfirmationMode confirmationMode)
     }
     if (confirm) {
         LogoutPromptIface iface;
-        iface.promptShutDown();
+        iface.promptShutDown().waitForFinished();
     } else {
         ShutdownIface iface;
-        iface.logoutAndShutdown();
+        iface.logoutAndShutdown().waitForFinished();
     }
 }
 
@@ -148,10 +148,10 @@ void SessionManagement::requestReboot(ConfirmationMode confirmationMode)
     }
     if (confirm) {
         LogoutPromptIface iface;
-        iface.promptReboot();
+        iface.promptReboot().waitForFinished();
     } else {
         ShutdownIface iface;
-        iface.logoutAndReboot();
+        iface.logoutAndReboot().waitForFinished();
     }
 }
 
@@ -166,10 +166,10 @@ void SessionManagement::requestLogout(ConfirmationMode confirmationMode)
     }
     if (confirm) {
         LogoutPromptIface iface;
-        iface.promptLogout();
+        iface.promptLogout().waitForFinished();
     } else {
         ShutdownIface iface;
-        iface.logout();
+        iface.logout().waitForFinished();
     }
 }
 
-- 
2.25.1

